diff --git a/Examples/PlatformBSP/OpenWRT_Generic_Linux/FileSystemInit.c b/Examples/PlatformBSP/OpenWRT_Generic_Linux/FileSystemInit.c
index 6ac2682..cd05d11 100644
--- a/Examples/PlatformBSP/OpenWRT_Generic_Linux/FileSystemInit.c
+++ b/Examples/PlatformBSP/OpenWRT_Generic_Linux/FileSystemInit.c
@@ -28,7 +28,7 @@ bool FileSystemInit = true;
 int fileSystemCreateRootFolders(void)
 {
 	int status = 0;
-	char folder[PAL_MAX_FILE_AND_FOLDER_LENGTH] = {0};
+	/*char folder[PAL_MAX_FILE_AND_FOLDER_LENGTH] = {0};
 
 	// Get default mount point.
 	status = pal_fsGetMountPoint(PAL_FS_PARTITION_PRIMARY, PAL_MAX_FILE_AND_FOLDER_LENGTH, folder);
@@ -68,6 +68,6 @@ int fileSystemCreateRootFolders(void)
         	printf("mkdir failed errno= %d\r\n",errno);
             return 1;
         }
-    }       
+    }*/       
 	return status;
 }
