diff --git a/qsdk/package/network/utils/mbed/mbed-cloud-client-example-restricted/mbed-cloud-client/update-client-hub/modules/pal-linux/scripts/openwrt/arm_update_kernel.sh b/qsdk/package/network/utils/mbed/mbed-cloud-client-example-restricted/mbed-cloud-client/update-client-hub/modules/pal-linux/scripts/openwrt/arm_update_kernel.sh
index 21d7837..cb80561 100755
--- a/qsdk/package/network/utils/mbed/mbed-cloud-client-example-restricted/mbed-cloud-client/update-client-hub/modules/pal-linux/scripts/openwrt/arm_update_kernel.sh
+++ b/qsdk/package/network/utils/mbed/mbed-cloud-client-example-restricted/mbed-cloud-client/update-client-hub/modules/pal-linux/scripts/openwrt/arm_update_kernel.sh
@@ -32,7 +32,7 @@
 KERNEL_PREFIX="ubi.mtd"
 
 # firmware prefix
-FIRMWARE_PREFIX="ROOT_"
+FIRMWARE_PREFIX="rootfs"
 
 # flags prefix
 FLAGS_PREFIX="FLAGS_"
@@ -51,45 +51,50 @@ for x in $(cat /proc/cmdline); do
 done
 
 # read active slot number
-ACTIVE_SLOT=${LINUX//${FIRMWARE_PREFIX}/}
+ACTIVE_SLOT=${LINUX#$FIRMWARE_PREFIX}
 
 # deduce alternate, update slot
-if [ $ACTIVE_SLOT == 0 ]; then
-    UPDATE_SLOT=1
+#if [ $ACTIVE_SLOT == "" ]; then
+if [ -z $ACTIVE_SLOT ]; then
+    UPDATE_SLOT="_1"
+	UPDATE_FLAG_SLOT="1"
+	ACTIVE_FLAG_SLOT="0"
 else
-    UPDATE_SLOT=0
+    UPDATE_SLOT=""
+	UPDATE_FLAG_SLOT="0"
+	ACTIVE_FLAG_SLOT="1"
 fi
 
 #
 # get active mtd for slot and flags
 #
 ACTIVE_NAME_MTD="${FIRMWARE_PREFIX}${ACTIVE_SLOT}"
-ACTIVE_LINE_MTD=$(cat /proc/mtd | grep ${ACTIVE_NAME_MTD})
+ACTIVE_LINE_MTD=$(cat /proc/mtd | grep -w ${ACTIVE_NAME_MTD})
 ACTIVE_SLOT_MTD=${ACTIVE_LINE_MTD%:*}
 
-ACTIVE_NAME_MTD="${FLAGS_PREFIX}${ACTIVE_SLOT}"
-ACTIVE_LINE_MTD=$(cat /proc/mtd | grep ${ACTIVE_NAME_MTD})
+ACTIVE_FLAGS_MTD="${FLAGS_PREFIX}${ACTIVE_FLAG_SLOT}"
+ACTIVE_LINE_MTD=$(cat /proc/mtd | grep -w ${ACTIVE_FLAGS_MTD})
 ACTIVE_FLAGS_MTD=${ACTIVE_LINE_MTD%:*}
 
 #
 # get update mtd for slot and flags
 #
 UPDATE_NAME_MTD="${FIRMWARE_PREFIX}${UPDATE_SLOT}"
-UPDATE_LINE_MTD=$(cat /proc/mtd | grep ${UPDATE_NAME_MTD})
+UPDATE_LINE_MTD=$(cat /proc/mtd | grep -w ${UPDATE_NAME_MTD})
 UPDATE_SLOT_MTD=${UPDATE_LINE_MTD%:*}
 
-UPDATE_NAME_MTD="${FLAGS_PREFIX}${UPDATE_SLOT}"
-UPDATE_LINE_MTD=$(cat /proc/mtd | grep ${UPDATE_NAME_MTD})
+UPDATE_NAME_MTD="${FLAGS_PREFIX}${UPDATE_FLAG_SLOT}"
+UPDATE_LINE_MTD=$(cat /proc/mtd | grep -w ${UPDATE_NAME_MTD})
 UPDATE_FLAGS_MTD=${UPDATE_LINE_MTD%:*}
 
 #
 # result
 #
-# echo "Active no.:   ${ACTIVE_SLOT}"
-# echo "Active slot:  ${ACTIVE_SLOT_MTD}"
-# echo "Active flags: ${ACTIVE_FLAGS_MTD}"
+#echo "Active no.:   ${ACTIVE_SLOT}"
+#echo "Active slot:  ${ACTIVE_SLOT_MTD}"
+#echo "Active flags: ${ACTIVE_FLAGS_MTD}"
 #
-# echo "Update no.:   ${UPDATE_SLOT}"
-# echo "Update slot:  ${UPDATE_SLOT_MTD}"
-# echo "Update flags: ${UPDATE_FLAGS_MTD}"
+#echo "Update no.:   ${UPDATE_SLOT}"
+#echo "Update slot:  ${UPDATE_SLOT_MTD}"
+#echo "Update flags: ${UPDATE_FLAGS_MTD}"
 #
